File: 04_counting_elements/missing_integer.py
def solution(A):
  n = len(A)
  lst = [0] * (n+2)
  for i in A:
    ind = max(i, 0)
    ind = min(n+1, ind)
    lst[ind] += 1
  for i in xrange(1, n+1):
    if lst[i] == 0:
      return i
  return n+1

File: 4_counting_elements/missing_integer.cpp
#include <vector>

using namespace std;

int solution(vector<int> &a) {
  int n = a.size();
  vector<int> cnts(n + 1, 0);
  for (int i = 0; i < n; i++) {
    if (0 < a[i] && a[i] < n + 1)
      cnts[a[i]] = 1;
  }
  for (int i = 1; i < n + 1; i++) {
    if (cnts[i] == 0)
      return i;
  }
  return n + 1;
}
